adcm_aio_client/core/objects/cm.py

```diff
@@ -374,3 +376,45 @@ async def _get_hosts_from_arg_or_filter(
             hosts = await self.filter(filters)  # type: ignore  # TODO
 
         return hosts
+
+
+class Job[Object: "InteractiveObject"](WithStatus, WithActions, RootInteractiveObject):
+    PATH_PREFIX = "jobs"
+
+    @property
+    def name(self: Self) -> str:
+        return str(self._data["name"])
+
+    @property
+    def start_time(self: Self) -> datetime:
+        return self._data["startTime"]
+
+    @property
+    def finish_time(self: Self) -> datetime:
+        return self._data["endTime"]
+
+    @property
+    def object(self: Self) -> Object:
+        obj_data = self._data["parentTask"]["objects"][0]
+        obj_type = obj_data["type"]
+
+        obj_dict = {
+            "host": Host,
+            "component": Component,
+            "provider": HostProvider,
+            "cluster": Cluster,
+            "service": Service,
+            "adcm": ADCM,
+        }
+
+        return self._construct(what=obj_dict[obj_type], from_data=obj_data)
+
+    @property
+    def action(self: Self) -> Action:
+        return self._construct(what=Action, from_data=self._data["parentTask"]["action"])
+
+    async def wait(self: Self, timeout: int = 30) -> None:
+        await asyncio.sleep(timeout)
```

Comment:
`wait` purpose is to wait for task / job to be finished (enter one of the final statuses)

so it shouldn't be just sleeping for given timeout

Reply:
changed